src/MOI_wrapper.jl

```diff
@@ -216,4 +216,89 @@ function backward!(model::Optimizer, params::Array{String}, dl_dz::Array{Float64
         end
     end
     return grads
-end
\ No newline at end of file
+end
+
+# MOI supports
+
+function MOI.supports(::Optimizer,
+    ::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}})
+    return true
+end
+
+function MOI.supports(::Optimizer,
+    ::MOI.ObjectiveFunction{MOI.ScalarQuadraticFunction{Float64}})
+    return true
+end
+
+function MOI.supports(::Optimizer, ::MOI.AbstractModelAttribute)
+    return true
+end
+
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarAffineFunction{Float64}}, ::Type{MOI.LessThan{Float64}}) = true
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarAffineFunction{Float64}}, ::Type{MOI.GreaterThan{Float64}}) = true
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarAffineFunction{Float64}}, ::Type{MOI.EqualTo{Float64}}) = true
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarQuadraticFunction{Float64}}, ::Type{MOI.LessThan{Float64}}) = true
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarQuadraticFunction{Float64}}, ::Type{MOI.GreaterThan{Float64}}) = true
+MOI.supports_constraint(::Optimizer, ::Type{MOI.ScalarQuadraticFunction{Float64}}, ::Type{MOI.EqualTo{Float64}}) = true
+
+
+MOI.get(model::Optimizer, ::MOI.SolveTime) = model.optimizer.solve_time
+
+function MOI.empty!(model::Optimizer)
+    MOI.empty!(model.optimizer)
+    model.primal_optimal = zeros(0)
```

Comment:
```suggestion
    model.primal_optimal .= 0
```
this avoids re-creating a new array, should be applied to each line here

Reply:
will do it